src/InstallerHosts/Elastic.InstallerHosts/Elasticsearch/Tasks/Immediate/ValidateArgumentsTask.cs

```diff
@@ -28,6 +29,11 @@ protected override bool ExecuteTask()
 				var validationFailures = ValidationFailures(failures);
 				throw new Exception("Cannot continue installation because of the following errors" + Environment.NewLine + validationFailures);
 			}
+
+			var installDir = this.Session.Get<string>("INSTALLDIR");
```

Comment:
`nameof(LocationsModel.InstallDir).ToUpperInvariant()`?

Reply:
The name of the property is not necessarily tied to the property name that gets emitted though. `INSTALLDIR` being a fixed default MSI property I think its better to leave as is.